stablehlo/dialect/TypeInference.cpp

```diff
@@ -283,6 +304,29 @@ LogicalResult verifyPairwiseCompatibleShapes(TypeRange values) {
   return success();
 }
 
+LogicalResult verifyTransposeOp(std::optional<Location> location,
+                                Type operandType, ArrayRef<int64_t> permutation,
+                                Type resultType) {
+  // transpose_c1
+  // part of transpose_c1 already verified by
+  // Trait HLO_CompatibleOperandsAndResultElementType
+  if (failed(isSameQuantScaleZeroPoint(location, operandType, resultType)))
```

Comment:
Transpose can be both per-axis and per-tensor. We are not matching the scales and zero-points for per-axis case?